src/components/date_picker/super_date_picker/super_date_picker.test.tsx

```diff
@@ -120,28 +120,81 @@ describe('EuiSuperDatePicker', () => {
     jest.useRealTimers();
   });
 
-  test('updateButtonProps', () => {
-    const updateButtonProps: EuiSuperDatePickerProps['updateButtonProps'] = {
-      fill: false,
-      color: 'ghost',
-    };
-
-    const component = mount(
-      <EuiSuperDatePicker
-        onTimeChange={noop}
-        updateButtonProps={updateButtonProps}
-      />
-    );
-    expect(component.find(EuiButton).props()).toMatchObject(updateButtonProps);
-  });
-
-  test('accepts data-test-subj and passes to EuiFormControlLayout', () => {
-    const component = shallow(
-      <EuiSuperDatePicker
-        onTimeChange={noop}
-        dataTestSubj="mySuperDatePicker"
-      />
-    );
-    expect(component).toMatchSnapshot();
+  describe('props', () => {
+    test('updateButtonProps', () => {
+      const updateButtonProps: EuiSuperDatePickerProps['updateButtonProps'] = {
+        fill: false,
+        color: 'ghost',
+      };
+
+      const component = mount(
+        <EuiSuperDatePicker
+          onTimeChange={noop}
+          updateButtonProps={updateButtonProps}
+        />
+      );
+      expect(component.find(EuiButton).props()).toMatchObject(
+        updateButtonProps
+      );
+    });
+
+    describe('showUpdateButton', () => {
+      test('can be false', () => {
+        const component = shallow(
+          <EuiSuperDatePicker onTimeChange={noop} showUpdateButton={false} />
+        );
+        expect(component).toMatchSnapshot();
+      });
+
+      test('can be iconOnly', () => {
+        const component = shallow(
+          <EuiSuperDatePicker onTimeChange={noop} showUpdateButton="iconOnly" />
+        );
+        expect(component).toMatchSnapshot();
+      });
+    });
+
+    test('accepts data-test-subj and passes to EuiFormControlLayout', () => {
+      const component = shallow(
+        <EuiSuperDatePicker
+          onTimeChange={noop}
+          data-test-subj="mySuperDatePicker"
+        />
+      );
+      expect(component).toMatchSnapshot();
+    });
+
+    describe('width', () => {
+      test('can be full', () => {
+        const component = shallow(
+          <EuiSuperDatePicker onTimeChange={noop} width="full" />
+        );
+        expect(component).toMatchSnapshot();
+      });
+      test('can be auto', () => {
+        const component = shallow(
+          <EuiSuperDatePicker onTimeChange={noop} width="auto" />
+        );
+        expect(component).toMatchSnapshot();
+      });
+    });
+
+    describe('compressed', () => {
+      test('is rendered', () => {
+        const component = shallow(
+          <EuiSuperDatePicker onTimeChange={noop} compressed />
+        );
+        expect(component).toMatchSnapshot();
+      });
+    });
```

Comment:
If this is a basic prop test without different conditions, I'd say you could just make this a `test()` without a parent describe (like `updateButtonProps` up above)
```suggestion
    test('is rendered', () => {
      const component = shallow(
        <EuiSuperDatePicker onTimeChange={noop} compressed />
      );
      expect(component).toMatchSnapshot();
    });
```

Also just wanted to say, I super appreciate the added tests + depth of testing conditions in this PR! 👏

Reply:
It is testing for the `compressed` prop. I tend to leave it with the `describe > test` format even if there's only a single `test` just for ease of scalability.

Reply:
Gotcha, that's fine